### PR TITLE
Clone in temporary directory

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/11836?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11836/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11836
```

</p>
</details>

### Related Issues/PRs

 #xxx

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: dev/check_patch_prs.py [hunk 1]
```diff
@@ -22,21 +23,32 @@ def get_commits(branch: str):
...
-            '--since="3 months ago"',
-            "--pretty=format:%H %s",
-        ],
-        text=True,
-    )
-    pr_rgx = re.compile(r"([a-z0-9]+) .+\s+\(#(\d+)\)$")
-    commits = []
-    for commit in log_stdout.splitlines():
-        if m := pr_rgx.search(commit.rstrip()):
-            commits.append(Commit(sha=m.group(1), pr_num=int(m.group(2))))
```
Comment (harupy) [inline dev/check_patch_prs.py:39]: The original implementation doesn't work if the branch is not fetched yet.

File: dev/check_patch_prs.py [hunk 2]
```diff
@@ -22,21 +23,32 @@ def get_commits(branch: str):
...
+                "--shallow-since=3 months ago",
+                "--branch",
+                branch,
+                "https://github.com/mlflow/mlflow.git",
+                tmpdir,
+            ],
+        )
+        log_stdout = subprocess.check_output(
+            [
+                "git",
```
Comment (harupy) [inline dev/check_patch_prs.py:40]: This is SLOW

Comment (harupy) [reply dev/check_patch_prs.py:40]: but this script is independent of the state of the git repo.

File: dev/check_patch_prs.py [hunk 3]
```diff
@@ -22,21 +23,32 @@ def get_commits(branch: str):
...
-    commits = []
-    for commit in log_stdout.splitlines():
-        if m := pr_rgx.search(commit.rstrip()):
-            commits.append(Commit(sha=m.group(1), pr_num=int(m.group(2))))
+    with tempfile.TemporaryDirectory() as tmpdir:
+        subprocess.check_call(
+            [
+                "git",
+                "clone",
+                "--shallow-since=3 months ago",
```
Comment (WeichenXu123) [inline dev/check_patch_prs.py:31]: Good setting!

**Review by WeichenXu123 — APPROVED** [review]
LGTM

**Pull request merged**